workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/ApplyTransformUpdatesSystem.cs

```diff
@@ -24,7 +25,13 @@ protected override void OnUpdate()
             for (var i = 0; i < transformUpdateData.Length; i++)
             {
                 var transformUpdates = transformUpdateData.TransformUpdate[i].Updates;
-                var lastTransformSnapshot = transformUpdateData.BufferedTransform[i].LastTransformSnapshot;
```

Comment:
What if there is nothing in the list and then we receive an update without all of the fields set? I don't think you can remove the `LastTransformSnapshot`.

Reply:
the `LastTransformSnapshot`initializes with the default `SpatialOSTransform`afaik, so the behaviour is exactly the same as before.

I agree that there are issues with this, but I don't think they are introduced in this PR.

Reply:
Though I can change the code in this system to work correctly even without `LastTransformSnapshot` :) will add a new commit

Reply:
Nice change :)